src/helpers/decorators/caching.decorator.ts

```diff
@@ -35,12 +36,36 @@ export function GetOrSetCache(cachingOptions: ICachingOptions) {
 
             const cachingService: CacheService = this.cachingService;
 
-            return await cachingService.getOrSet(
+            let cachedValue = await cachingService.get(cacheKey);
+
+            if (cachedValue !== undefined) {
+                cachedValue = cachedValue === 'null' ? null : cachedValue;
+                cachedValue =
+                    cachedValue === 'undefined' ? undefined : cachedValue;
+
+                return cachedValue;
+            }
+
+            const value = await originalMethod.apply(this, args);
+
+            let { remoteTtl, localTtl } = cachingOptions;
+
+            if (typeof value === 'undefined' || value === null) {
+                remoteTtl = CacheTtlInfo.NullValue.remoteTtl;
+                localTtl = CacheTtlInfo.NullValue.localTtl;
+            }
+
+            cachedValue = typeof value === 'undefined' ? 'undefined' : value;
```

Comment:
duplicated code. maybe extract a function for returning the cached value in case of 'null' or 'undefined'.